### ISSUE TITLE
Directories with spaces don't get updated correctly

### ISSUE THREAD

**Issue by guillaumeboehm** [agkozak/zsh-z#61]
It seems that jumping into a directory containing spaces adds it to the database but doesn't update it if it is already in the database.

**Comment by agkozak** [comment]
You're absolutely right. I'll fix this soon. Thanks!

**Comment by agkozak** [comment]
@guillaumeboehm,

I just pushed a fix to the `develop` branch -- would you give that a pull and see if it works for you? Thanks.

**Comment by agkozak** [comment]
Don't worry - I think there may be a few related problems that I'll want to fix - I'll be happy to make the changes.

But could you give me an example of the "match search" problem? If I type `z foo\ bar`, it does take me to `~/foo\ bar`.

**Comment by guillaumeboehm** [comment]
You're right... I'm testing it right now and can't reproduce it either... I don't even know what I was typing when I tested it. Never mind it lol.
